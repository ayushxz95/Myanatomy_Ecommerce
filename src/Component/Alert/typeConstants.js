export const mockItem = {
      state:5,
      id:1,
      type:"Electronics",
      statetype:"Electronics",
      Alertmessage:"Cancel",
      color:"error",
  };