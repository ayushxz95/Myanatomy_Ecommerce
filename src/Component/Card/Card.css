:root {
  --clr-gray-light: #d7dfe2;
  --clr-gray-med: #616b74;
  --clr-gray-dark: #414b56;
  --clr-link: #4d97b2;
  --clr-popular: #ef257a;
  --clr-technology: #651fff;
  --clr-psychology: #e85808;
}

* {
  box-sizing: border-box;
  font-family: "Inter", sans-serif;
  margin: 0;
  padding: 0;
}

body {
  /* margin: 2rem; */
  color: var(--clr-gray-dark);
}
.button-holder {
  float: right;
  /* border-color: black; */
  align-items: center;
}
/* .button-section {
    width: 50%;
  }
  
  .buttonssection-button{
    border: none;
    outline: none;
    padding: 12px;
    margin: 0 auto;
    text-align: center;
    cursor: pointer;
    width: 70%;
    font-size: 12px;
    font-weight: bold;
    background-color: white;
    padding: 1em;
  }
  .buttonssection-button :hover{
    color:#616b74;
  } */

.wrapper {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.card {
  overflow: hidden;
  box-shadow: 0px 2px 20px var(--clr-gray-light);
  background: white;
  border-radius: 0.5rem;
  position: relative;
  width: 250px;
  margin: 3.2rem;
  cursor: pointer;
}
.card-body {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.card:hover {
  box-shadow: 0px 2px 40px var(--clr-gray-light);
}

.banner-img {
  position: absolute;
  object-fit: cover;
  height: 14rem;
  width: 100%;
}

.category-tag {
  font-size: 0.8rem;
  font-weight: bold;
  color: white;
  background: red;
  padding: 0.5rem 1.3rem 0.5rem 1rem;
  text-transform: uppercase;
  position: absolute;
  z-index: 1;
  top: 1rem;
  border-radius: 0 2rem 2rem 0;
}
.delete-tag {
  font-size: 0.8rem;
  font-weight: bold;
  color: white;
  background: red;
  padding: 0.5rem 1.3rem 0.5rem 1rem;
  text-transform: uppercase;
  position: absolute;
  z-index: 1;
  top: 1rem;
  right: 0rem;
  border-radius: 2rem 0rem 0rem 2rem;
}
.delete-button {
  border: none;
  background: transparent;
}
.delete-button:hover {
  cursor: pointer;
}

.popular {
  background: var(--clr-popular);
}

.technology {
  background: var(--clr-technology);
}

.psychology {
  background: var(--clr-psychology);
}

.card-body {
  margin: 15rem 1rem 1rem 1rem;
}

*/ .card-title {
  font-size: small;
  line-height: 1.5rem;
  margin: 1rem 0 0.5rem;
}

.card-description {
  color: var(--clr-gray-med);
  font-size: 0.9rem;
}

.card-profile {
  display: flex;
  margin-top: 2rem;
  align-items: center;
}

.profile-img {
  width: 60px;
  height: 60px;
  object-fit: cover;
  border-radius: 50%;
}

.card-profile-info {
  margin-left: 1rem;
}

.profile-name {
  font-size: 1rem;
}

.profile-followers {
  color: var(--clr-gray-med);
  font-size: 0.9rem;
}

.card-title {
  color: #ef257a;
  font-size: large;
  white-space: nowrap;
  width: 140px;
  overflow: hidden;
  text-overflow: ellipsis;
}
.company-name {
  font-size: small;
  white-space: nowrap;
  width: 100px;
  overflow: hidden;
  text-overflow: ellipsis;
  margin-top: 5px;
}
