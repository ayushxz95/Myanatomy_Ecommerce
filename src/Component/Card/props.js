import { deleteItem } from "../../Redux/Actions/ItemAction";

const mapDispatchToProps = dispatch => {
    return {
        deleteItem: (id) => dispatch(deleteItem(id))
    }
}

export {mapDispatchToProps};