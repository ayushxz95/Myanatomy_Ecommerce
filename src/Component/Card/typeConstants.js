export const mockItem = {
    name: "Fan",
    id: 1,
    type: "Electronics",
    company: "Havells",
    key: 1,
  };