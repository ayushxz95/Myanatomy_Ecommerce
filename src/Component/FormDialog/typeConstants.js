export const mockItem = {
  message: "Update Product",
  buttonmessage: "Update",
  type: "Electronics",
  company: "Bajaj",
  name: "Fan",
  id: 3,
};
