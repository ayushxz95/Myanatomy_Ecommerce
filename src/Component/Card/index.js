import { connect } from "react-redux";
import Card from "./Card";
import {mapDispatchToProps} from "./props";

export default connect(null, mapDispatchToProps)(Card);