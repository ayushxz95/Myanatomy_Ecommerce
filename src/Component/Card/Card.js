import React, { Component } from "react";
import { AiFillDelete } from "react-icons/ai";
import { PropTypes } from "prop-types";
import "./Card.css";
import FormDialog from "../FormDialog/index";


class Card extends Component {
  handleDelete = () => {
      this.props.deleteItem(this.props.id, this.props.category);
  }
  render() {
    const { name, type, company, id, image } = this.props;
    // if(type === 'Cloths')
    // var image = "https://images.unsplash.com/photo-1523381210434-271e8be1f52b?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2070&q=80";
    // else if(type === 'Electronics')
    // var image = "https://images.unsplash.com/photo-1515879218367-8466d910aaa4?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&ixid=eyJhcHBfaWQiOjE0NTg5fQ";
    // else if(type === 'Grocery')
    // var image = "https://images.unsplash.com/photo-1542838132-92c53300491e?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=774&q=80";
    return (
      <div className="card">
        <div className="card-banner">
          <p className="category-tag popular">{type}</p>
          <p className="delete-tag popular">
            <button
            className="delete-button"
                onClick={() => {this.handleDelete()}}
                title="Delete"
                id="btn-card33"
              >
                <AiFillDelete size="14px" color="white" />
              </button></p>
          <img
            className="banner-img"
            src={image}
            alt=""
          />
        </div>
        <div className="card-body">
          <div>
          <h2 className="card-title">{name}</h2>
          <h4 className="company-name">{company}</h4>
          </div>
          <div className="button-holder">
              <FormDialog 
              message={"Update Product"}
              buttonmessage={"Update"}
              type={type}
              company={company}
              name={name}
              image={image}
              id={id}
              />
          </div>
        </div>
      </div>
    );
  }
}

Card.propTypes = {
    deleteItem: PropTypes.func
}

export default Card;