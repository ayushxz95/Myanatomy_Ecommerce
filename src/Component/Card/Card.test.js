import { shallow, configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Card from "./Card";
import {mockItem} from "./typeConstants"

configure({ adapter: new Adapter() });
// let component;

describe("Testing Card Component", () => {
  it("Matching Card's snapshot", () => {
    const deleteItemmock = jest.fn();
    // console component what shallow will return and mount will return
    // naming of snapshot
    const component = shallow(
      <Card
        deleteItem={deleteItemmock}
        id={mockItem.id}
        name={mockItem.name}
        type={mockItem.type}
        company={mockItem.company}
        key={mockItem.key}
      />
    );
    //console.log(component);
    expect(component).toMatchSnapshot();
  });
  it("Name is present in props", () => {
    const deleteItemmock = jest.fn();
    const component = shallow(
      <Card
        deleteItem={deleteItemmock}
        id={mockItem.id}
        name={mockItem.name}
        type={mockItem.type}
        company={mockItem.company}
        key={mockItem.key}
      
      />
    );
    const name_prop = component.find(".card-title")
    expect(name_prop.text()).toBe("Fan");
  });
  it("Company is present in props", () => {
    const deleteItemmock = jest.fn();
    const component = shallow(
      <Card
        deleteItem={deleteItemmock}
        id={mockItem.id}
        name={mockItem.name}
        type={mockItem.type}
        company={mockItem.company}
        key={mockItem.key}
  
      />
    );
    const Companyname_prop = component.find(".company-name")
    expect(Companyname_prop.text()).toBe("Havells");
  });
  it("Button is Clickable", () => {
    const deleteItemmock = jest.fn();
    const component = shallow(
      <Card
        deleteItem={deleteItemmock}
        id={mockItem.id}
        name={mockItem.name}
        type={mockItem.type}
        company={mockItem.company}
        key={mockItem.key}
      />
    );
    const instance = component.instance();
    console.log(instance); // spyOn and mocking diff
    jest.spyOn(instance, "handleDelete"); 
    // jest.spyOn(object, methodName).mockImplementation(() => customImplementation)
    const deleteButton = component.find("#btn-card33");
    deleteButton.simulate("click");
    expect(deleteItemmock).toHaveBeenCalledTimes(1);
  });
});
