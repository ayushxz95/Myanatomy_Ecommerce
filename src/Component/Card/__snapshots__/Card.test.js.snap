// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Card Component Matching Card's snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-banner"
  >
    <p
      className="category-tag popular"
    >
      Electronics
    </p>
    <p
      className="delete-tag popular"
    >
      <button
        className="delete-button"
        id="btn-card33"
        onClick={[Function]}
        title="Delete"
      >
        <AiFillDelete
          color="white"
          size="14px"
        />
      </button>
    </p>
    <img
      alt=""
      className="banner-img"
    />
  </div>
  <div
    className="card-body"
  >
    <div>
      <h2
        className="card-title"
      >
        Fan
      </h2>
      <h4
        className="company-name"
      >
        Havells
      </h4>
    </div>
    <div
      className="button-holder"
    >
      <Connect(FormDialog)
        buttonmessage="Update"
        company="Havells"
        id={1}
        message="Update Product"
        name="Fan"
        type="Electronics"
      />
    </div>
  </div>
</div>
`;
